### ISSUE TITLE
NPE loading credentials.json if the main class isn't in the root package

### ISSUE THREAD

**Issue by GavinC1985** [gsuitedevs/java-samples#14]
(Please fill out these details before submitting an issue)

### Sample Name
(i.e. Google Vault API Quickstart)
https://developers.google.com/vault/quickstart/java

### Expected Behavior
List the first 10 matters on your account. 

### Actual Behavior
It looks like the app is failing to find the credentials.json file and throws an exception:

Exception in thread "main" java.lang.NullPointerException
	at java.io.Reader.<init>(Reader.java:78)
	at java.io.InputStreamReader.<init>(InputStreamReader.java:72)
	at main.java.Main.authorize(Main.java:71)
	at main.java.Main.getVaultService(Main.java:95)
	at main.java.Main.main(Main.java:103)

I have actually tried a number of your java samples that have a similar workflow to this and they are returning the same error. The Drive API- https://developers.google.com/drive/api/v3/quickstart/java
is giving me the same error. 

I have tried updating the modules to their latest versions via maven but it has had no effect. 

'com.google.api-client:google-api-client:1.24.1'
'com.google.oauth-client:google-oauth-client-jetty:1.24.1'

Any help would be greatly appreciated.

### Specifications
  - Java version (`java -version`)
Java version - 1.8.0_131 

- OS (Mac/Linux/Windows)
Mac - version 10.12.6

**Comment by GavinC1985** [comment]
Aha okay I figured it out - it was the 
InputStream in = Main.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
This should be  InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH);
The credentials path being the full path:
 private static final String CREDENTIALS_FILE_PATH = "src/main/resources/credentials.json";


Hmm it looks like this reference is stale in a lot of the articles...

  InputStream in =Quickstart.class.getResourceAsStream("/credentials.json");

Anyway that fixed it for me :-D

**Comment by mlizbeth** [comment]
As long as you add a "/" to CREDENTIALS_FILE_PATH (and assuming credentials is in src/main/java which is where "mkdir -p src/main/java src/main/resources" comes in.

mkdir --parents src/main/java src/main/resources, which if I am interpreting the man pages correctly, it's kind of like a symbolic link, where your resources files should be visible from src/main/java. BUT you still need the path as "/credentials.json" because src/main/java is the root.

tl;dr, CREDENTIALS_FILE_PATH = "/credentials.json" will suffice

**Comment by erickoledadevrel** [comment]
Thanks @mlizbeth for the tip, it seems like there is an issue if the class is not in the root package, since the credentials.json is currently being loaded with a relative path. The instructions say to put the class in the root package, but it's less error prone to specify the absolute path. I'll send out a PR.

@GavinC1985 - It looks like you aren't using Gradle as specified in the instructions, and you are setting the source root at `/src` instead `/src/main/java`, resulting in incorrect packages like `main.java.Main`.
